### PR TITLE
Source zendesk support: validate datetime on check

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/1683

## How
Add validation of a `start_date` param when checking connectivity

### CODE REVIEW

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/source.py [hunk 1]
```diff
@@ -82,8 +82,9 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
         auth = self.get_authenticator(config)
         settings = None
         try:
+            datetime.strptime(config["start_date"], DATETIME_FORMAT)
             settings = UserSettingsStream(config["subdomain"], authenticator=auth, start_date=None).get_settings()
-        except requests.exceptions.RequestException as e:
+        except Exception as e:
```
Comment (bnchrch) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/source.py:87]: Could we be slightly more restrictive here and catch a datetime format issue in addition to the request exception?

To catch all exceptions could lead to some silent issues for customers.

**Review by bnchrch — APPROVED** [review]
Nice work. 👍 

One comment I do think we should address.

I will approve so as not to be a blocker.

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/source.py (hunk 1 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/source.py:87]: @bnchrch this is done on purpose. We want every exception to be caught, I would say this is a pattern for the `check` command implementation

**Pull request merged**